src/redis-cli.c

```diff
@@ -9390,7 +9413,7 @@ static void statMode(void) {
 
         /* Keys */
         aux = 0;
-        for (j = 0; j < 20; j++) {
+        for (j = 0; j < dbnum; j++) {
```

Comment:
Why not just loop up to `j < INT_MAX` and `break` when getLongInfoField fails?

No need for an extra roundtrip to redis to query the number of dbs.

Reply:
> break when getLongInfoField fails?

ohh, it doesn't seem to work

```
# Keyspace
db0:keys=1,expires=0,avg_ttl=0
db10:keys=1,expires=0,avg_ttl=0
```

Reply:
What do you mean? If db 1..9 have no keys, they are not listed in INFO?

Reply:
>  If db 1..9 have no keys, they are not listed in INFO?

yes

Reply:
Ah, you're right. I can see in server.c it is skipped if the db has no keys:

            if (keys || vkeys) {
                info = sdscatprintf(info,
                    "db%d:keys=%lld,expires=%lld,avg_ttl=%lld\r\n",
                    j, keys, vkeys, server.db[j].avg_ttl);
            }

So I guess we must use CONFIG GET.